source/frontend/src/features/leases/add/AddLeaseContainer.tsx

```diff
@@ -71,6 +71,10 @@ export const AddLeaseContainer: React.FunctionComponent<
   };
 
   const handleSave = () => {
+    if (formikRef?.current?.isValid) {
+      setIsMissingFields(false);
```

Comment:
this pattern works, but minor nitpick. Validation will fail for a number of reasons, invalid fields, missing fields, fields that are too long, etc. variable could be renamed to something more generic such as isFormInvalid.